plugins/modules/database/saphana/hana_query.py

```diff
@@ -0,0 +1,187 @@
+    return list(reader)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            sid=dict(type='str', required=True),
+            instance=dict(type='str', required=True),
+            encrypted=dict(type='bool', required=False, default=False),
+            host=dict(type='str', required=False),
+            user=dict(type='str', required=False, default="SYSTEM"),
+            password=dict(type='str', required=True, no_log=True),
+            database=dict(type='str', required=False),
+            query=dict(type='list', elements='str', required=False),
+            filepath=dict(type='list', elements='path', required=False),
+            autocommit=dict(type='bool', required=False, default=True),
+        ),
+        required_one_of=[('query', 'filepath')],
+        supports_check_mode=False,
+    )
+    rc, out, err, out_raw = [0, [], "", ""]
+
+    params = module.params
+
+    sid = (params['sid']).upper()
+    instance = params['instance']
+    user = params['user']
+    password = params['password']
+    autocommit = params['autocommit']
+    host = params['host']
+    database = params['database']
+    encrypted = params['encrypted']
+
+    filepath = params['filepath']
+    query = params['query']
+
+    bin_path = "/usr/sap/{sid}/HDB{instance}/exe/hdbsql".format(sid=sid, instance=instance)
+
+    try:
+        command = [module.get_bin_path(bin_path, required=True)]
+    except Exception as e:
+        module.fail_json(msg='Failed to find hdbsql at the expected path "{0}". Please check SID and instance number: "{1}"'.format(bin_path, to_native(e)))
+
+    if encrypted is True:
+        command.extend(['-attemptencrypt'])
+    if autocommit is False:
+        command.extend(['-z'])
+    if host is not None:
+        command.extend(['-n', host])
+    if database is not None:
+        command.extend(['-d', database])
+    # -x Suppresses additional output, such as the number of selected rows in a result set.
+    command.extend(['-x', '-i', instance, '-u', user, '-p', password])
+
+    if filepath is not None:
+        command.extend(['-I'])
+        for p in filepath:
+            # makes a command like hdbsql -i 01 -u SYSTEM -p secret123# -I /tmp/HANA_CPU_UtilizationPerCore_2.00.020+.txt,
+            # iterates through files and append the output to var out.
+            query_command = command + [p]
+            (rc, out_raw, err) = module.run_command(query_command)
+            out.append(csv_to_list(out_raw))
+    if query is not None:
+        for q in query:
+            # makes a command like hdbsql -i 01 -u SYSTEM -p secret123# "select user_name from users",
+            # iterates through multiple commands and append the output to var out.
+            query_command = command + [q]
+            (rc, out_raw, err) = module.run_command(query_command)
+            out.append(csv_to_list(out_raw))
+    changed = True
+
+    module.exit_json(changed=changed, message=rc, query_result=out, stderr=err)
```

Comment:
```suggestion
    module.exit_json(changed=changed, rc=rc, query_result=out, stderr=err)
```
I would expect a string for `message`, not an integer.